vm_data.py-dist

```diff
@@ -27,3 +53,42 @@ VMS = {
         "uefi_windows": [],
     }
 }
+
+from fnmatch import fnmatch
+
+# Prepend the XVA_LOCATION to the VM filename
+# Returns None if a host version is passed with host_version and matches the one
+# specified with the vm filename, if any
+def prepend_xva_location(vm, host_version):
+    if type(vm) is dict:
+        if len(vm) != 1:
+            return None
+
+        xva = list(vm)[0]
+        if host_version is None or fnmatch(host_version, vm[xva]):
+            return XVA_LOCATION + xva
+
+        return None
+
+    return XVA_LOCATION + vm
+
+# Returns a new dict based on the VMS dict declared above. It is mostly
+# duplicated but some entries could be removed with host version checking.
+# (It is possible to remove items from a dict/list while iterating it but that
+# can become fragile and not as clear as creating a new dict)
+def get_vms(host_version):
+    res = dict({"single": dict(), "multi": dict()})
+
+    for key, vm in VMS["single"].items():
+        xva = prepend_xva_location(vm, host_version)
+        if xva is not None:
+            res["single"][key] = xva
+
+    for key, vms in VMS["multi"].items():
+        res["multi"][key] = list()
+        for vm in vms:
+            xva = prepend_xva_location(vm, host_version)
+            if xva is not None:
+                res["multi"][key].append(xva)
+
+    return res
```

Comment:
I'd rather have all this code directly in jobs.py, so that vm_data.py remains just about data.